homeassistant/components/mqtt/climate.py

```diff
@@ -405,9 +406,15 @@ def render_template(msg, template_name):
         def handle_action_received(msg):
             """Handle receiving action via MQTT."""
             payload = render_template(msg, CONF_ACTION_TEMPLATE)
-
-            self._action = payload
-            self.async_write_ha_state()
+            if payload is None or payload in CURRENT_HVAC_ACTIONS:
```

Comment:
First of all, it's not clear from the documentation that the action should be set to `None`, I've opened a PR to clarify that.

We probably also need a special payload which will be interpreted as `None` when templates are not used, if it's allowed to set the action to `None`. This is used in MQTT fan to reset the percentage to `None`: https://www.home-assistant.io/integrations/fan.mqtt/#payload_reset_percentage

Reply:
I basically just added None there based on @bdraco's comment. Setting None via async_fire_mqtt_message doesn't work apparently(?).

Anyway, based on the doc, would it make sense to have the unit test run something along the lines of
`assert all(elem in actions  for elem in CURRENT_HVAC_ACTIONS)` in order to catch future edits of CURRENT_HVAC_ACTIONS?

Reply:
It's not possible to receive `None` directly over MQTT, an empty MQTT payload will be interpreted as the empty string.
If it's necessary to allow resetting the action to `None`, we should enable that functionality in the same way as for MQTT fan.
If it's not necessary, drop the None-check.

Your idea for a test seems reasonable.

Reply:
tbh I am not sure None is a valid action that can be set given the description of HVAC action. I suppose None really just makes sense as the initial value to be interpreted as "unknown" - which no further update should result in? Unless a device is paired/reset again maybe.
I removed the None check for now given your remarks and added the assert all

Reply:
For clarity

The explicit `None` is a valid value `hvac_action` 
The string `"None"` is not a valid value for `hvac_action`. (which is what was seen in the HomeKit PR)

Reply:
So, should I add the None check/allowance back in?

Reply:
General question is should we equate "None" with None? In current change "None" will log error message.

Reply:
> The general question is: should we equate "None" with None? In a current change "None" will log error message.

I guess that goes hand in hand with accepting None as a valid type to be used for hvac action to reset it (assuming "None", "none", "null" etc should be interpreted as None) - which @emontnemery mentioned above should happen using a special payload

Reply:
In my mind, if we allow `None` as a HVAC action (or perhaps rather the absence of a HVAC action) , we should:
- Update the developer docs to make that clear in the same way as it's made clear for fan https://www.home-assistant.io/integrations/fan.mqtt/#payload_reset_percentage
- Make it possible to configure the "None" payload for MQTT Climate, it can perhaps default to `"None"`.

There's a somewhat related discussion about Fan, about no longer supporting `None`: https://github.com/home-assistant/core/pull/59688

Reply:
For reference, https://www.home-assistant.io/integrations/climate.mqtt#action_topic currently only lists the 6 valid values checked in this PR.
Based on @bdraco's comment above, we could default to `None` (not `"None"`) upon receiving an invalid value since there is _some_ unknown state not handled by HA in place until further notice (similar to what the state within HA is upon startup, thus being a valid case?).
However, any side-effects of doing so are sorta unknown I guess? A warning should be logged in that case imho

Reply:
If `None` isn't actually supported now there is no need to add it.  It would also be perfectly reasonable to reject invalid values instead of setting them to `None`

Reply:
So, how should we proceed? Take this PR as the bugfix that it is intended to be and not accept values not documented or extend it to allow some variation of `None`?

Reply:
I think None is only valid as an initial state. There is no reason to reset it later e.g. if the device becomes unavailable. This is different from the MQTT fan where speed percentage and preset_mode state can become invalid, e.g. when a fan has one mode `auto`, and setting a fixed speed percentage will reset this mode.

Further: The value template will return `None` if a value not could be parsed, we should not use this for other means. I think this PR is fine, but I would log a warning, instead of an error.